### PR TITLE
Features/rhpm cd11

### PR DESCRIPTION
**Pull request by karthikkharvi**
RHPM_CD11 Api for download candidate resume

### CODE REVIEW

File: api/RS.API/RS.Web/Controllers/CandidateController.cs [hunk 1]
```diff
@@ -116,5 +120,22 @@ private void GetCandidateDocumentDetails(CandidateDocumentViewModel candidateDoc
 
         }
 
+        [HttpPost]
+        public ActionResult DownloadFile(String file)
```
Comment (salilsaini868) [inline api/RS.API/RS.Web/Controllers/CandidateController.cs]: Purpose of using String ?

File: api/RS.API/RS.Web/Controllers/CandidateController.cs [hunk 2]
```diff
@@ -116,5 +120,22 @@ private void GetCandidateDocumentDetails(CandidateDocumentViewModel candidateDoc
...
+        {
+            if (file == null)
+            {
+                return null;
+            }
+            var path = Path.Combine(_hostingEnvironment.ContentRootPath, "uploadFiles\\"+ file);
```
Comment (salilsaini868) [inline api/RS.API/RS.Web/Controllers/CandidateController.cs]: add static string in json file

File: web/src/app/candidate/candidate.component.html [hunk 3]
```diff
@@ -137,8 +137,8 @@
                     <div>
                         <input type="file" (change)="fileChange($event)" placeholder="Upload file" accept=".pdf,.doc,.docx">
                     </div>
-                    <div  *ngIf="candidateModel.candidateId != null">
-                        <a (click) = "downloadFile()">{{ candidateModel.candidateDocument.documentName}}</a>
+                    <div *ngIf="candidateModel.candidateId != null">
+                        <a (click)="downloadFile(candidateModel.candidateDocument.uploadedDocument, candidateModel.candidateDocument.documentName)" download="candidateModel.candidateDocument.documentName">{{ candidateModel.candidateDocument.documentName}}</a>
```
Comment (salilsaini868) [inline web/src/app/candidate/candidate.component.html]: change method name

File: web/src/app/candidate/candidateDetails.component.html [hunk 4]
```diff
@@ -20,7 +20,11 @@
                             <label>{{ 'CANDIDATE.FIRSTNAME' | translate }}</label>
                         </div>
                         <div class="col-md-9 col-sm-9 col-xs-12">
-                            <output name="firstName">{{ candidateModel.firstName }}</output>
+                            <output class="pull-left" name="firstName">{{ candidateModel.firstName }}</output>
+                            <div class="pull-right down-arrow" *ngIf="candidateModel.candidateId != null">
+                                <a (click)="downloadFile(candidateModel.candidateDocument.uploadedDocument, candidateModel.candidateDocument.documentName)"
+                                    download="candidateModel.candidateDocument.documentName"><i title="Download Candidate Resume" class="fa fa-download"></i></a>
```
Comment (salilsaini868) [inline web/src/app/candidate/candidateDetails.component.html]: title should come from resource files

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

File: api/RS.API/RS.Web/Controllers/CandidateController.cs (hunk 1 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Web/Controllers/CandidateController.cs]: Done

File: web/src/app/candidate/candidate.component.html (hunk 3 above)
Comment (karthikkharvi) [reply web/src/app/candidate/candidate.component.html]: Done

File: web/src/app/candidate/candidateDetails.component.html (hunk 4 above)
Comment (karthikkharvi) [reply web/src/app/candidate/candidateDetails.component.html]: Done

File: api/RS.API/RS.Web/Controllers/CandidateController.cs (hunk 2 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Web/Controllers/CandidateController.cs]: Done

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**